ros2lifecycle/test/test_cli.py

```diff
@@ -0,0 +1,342 @@
+
+
+class TestROS2LifecycleCLI(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(
+        cls,
+        launch_service,
+        proc_info,
+        proc_output,
+        rmw_implementation
+    ):
+        @contextlib.contextmanager
+        def launch_lifecycle_command(self, arguments):
+            lifecycle_command_action = ExecuteProcess(
+                cmd=['ros2', 'lifecycle', *arguments],
+                additional_env={
+                    'RMW_IMPLEMENTATION': rmw_implementation,
+                    'PYTHONUNBUFFERED': '1'
+                },
+                name='ros2lifecycle-cli',
+                output='screen'
+            )
+            with launch_testing.tools.launch_process(
+                launch_service, lifecycle_command_action, proc_info, proc_output,
+                output_filter=launch_testing_ros.tools.basic_output_filter(
+                    filtered_rmw_implementation=rmw_implementation
+                )
+            ) as lifecycle_command:
+                yield lifecycle_command
+        cls.launch_lifecycle_command = launch_lifecycle_command
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_list_lifecycle_node_transitions(self):
+        with self.launch_lifecycle_command(
+            arguments=['list', 'lc_node']
+        ) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=[
+                '- configure [1]',
+                '\tStart: unconfigured',
+                '\tGoal: configuring',
+                '- shutdown [5]',
+                '\tStart: unconfigured',
+                '\tGoal: shuttingdown'
+            ],
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_list_all_lifecycle_node_transitions(self):
+        with self.launch_lifecycle_command(
+            arguments=['list', 'lc_node', '-a']
+        ) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=ALL_LIFECYCLE_NODE_TRANSITIONS,
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_list_all_lifecycle_hidden_node_transitions(self):
+        with self.launch_lifecycle_command(
+            arguments=['list', '_hidden_lc_node', '-a']
+        ) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=ALL_LIFECYCLE_NODE_TRANSITIONS,
```

Comment:
This should not return any transition since you didn't give the `--include-hidden-nodes` argument

Reply:
That's true, which makes me wonder how's this test passing. Looks like `ros2 lifecycle list` is simply ignoring the `--include-hidden-nodes` flag.

Reply:
See #395.

Reply:
Test fixed in https://github.com/ros2/ros2cli/pull/391/commits/fc5977ccf05dac4a6800cda1175f95335b249ebb.

Reply:
Can you make the same tests but without the hidden flag? To ensure the proper functionality of the flag (I'm seeing that it will work, but just in case)

Reply:
Done in https://github.com/ros2/ros2cli/pull/391/commits/eb2f8d38f413ebd74fa7d91b1019e84487ba00ec